crates/node/src/catchup.rs

```diff
@@ -35,8 +36,11 @@ impl Node {
         };
 
         let request = match from_json_slice(&message?.data)? {
-            CatchupStreamMessage::Request(req) => req,
+            CatchupStreamMessage::TransactionsRequest(req) => req,
             message @ (CatchupStreamMessage::TransactionsBatch(_)
+            | CatchupStreamMessage::ApplicationBlobRequest(_)
+            | CatchupStreamMessage::ApplicationBlobSize(_)
```

Comment:
all peers already know this information, following #719, the client peer has that information too

`is_application_installed` returns `false` in either of these cases;
- the application is unavailable
- the blob of the application is unavailable

But where `is_application_installed` fails, you can still attempt to fetch the application via `ContextManager::get_application` to determine if it's just the blob that unavailable, and you can proceed to request it from your peer

Reply:
Following the considerations and accommodations made in https://github.com/calimero-network/core/pull/714, I'll be adding blob size to the contract in a coming PR, as we cannot trust the size the peer gives us either.